smaily_for_drupal/src/Form/SmailyForDrupalAdminForm.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+namespace Drupal\smaily_for_drupal\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+
+/**
+ * Smaily module admin configuraiton form.
+ */
+class SmailyForDrupalAdminForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'smaily_for_drupal.adminsettings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'smaily_for_drupal_admin_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('smaily_for_drupal.adminsettings');
+
+    // Wrapper for editing and returning form data with ajax.
+    $form['smaily_container'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'id' => 'smaily-wrapper',
+      ],
+    ];
+
+    $form['smaily_container']['credentials'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('API credentials'),
+    ];
+
+    $form['smaily_container']['credentials']['message'] = [
+      '#type' => 'markup',
+      '#markup' => '<div class="smaily_message"></div>',
+    ];
+
+    $form['smaily_container']['credentials']['smaily_domain'] = [
```

Comment:
I would go with `['smaily_api_credentials']['domain']` (and similar) grouping. It makes more sense.